pytest_fixtures/core/contenthosts.py

```diff
@@ -350,3 +351,115 @@ def custom_host(request):
+    # delete the updated repositories
+    for repo in repositories.values():
+        repo.read().delete()
+
+
+@pytest.fixture
+def registered_contenthosts(request, target_sat, module_org, module_ak, module_cv, fake_yum_repos):
+    """Create and register multiple contenthosts to satellite.
+    Parametrize int `count` (num of registered hosts), default: 2.
+        by passing a request containing param 'count'.
+    Parametrize int `rhel` (deploy_rhel_version), default: DEFAULT_RHEL_VERSION.
+        by passing a request containing param 'rhel'.
+    Parametrize str `workflow` (used for checkout), default: 'deploy-template'.
+        by passing a request containing param 'workflow'.
+
+    example: No parametrization will result in 2 hosts of DEFAULT RHEL Version.
+
+    eg. distro=rhel10, count=5, default workflow:
+        @pytest.mark.parametrize(
+            'registered_contenthosts',
+            [{'rhel': 10, 'count': 5}],
+            indirect=True,
+        )
+
+    Note: Does Not make use of `markers.rhel_ver_match`,
+        or `markers.rhel_ver_list`, leads to duplicate parametrizations.
+    Three custom repositories will be synced & enabled initially,
+    Add, enable, and sync any other repos to CV/AK for hosts, following fixture setup.
+    """
+    _config = host_conf(request)
+    param = getattr(request, 'param', {})
+    count = param.get('count', 2)
+    assert all([isinstance(count, int), count > 0]), 'Count must be an integer greater than zero.'
+
+    _config['deploy_rhel_version'] = str(
+        param.get('rhel', settings.content_host.default_rhel_version)
+    )
+    _config['distro'] = 'rhel'
+    _config['rhel_version'] = _config['deploy_rhel_version']
+    _config['workflow'] = param.get('workflow', 'deploy-template')
+    assert isinstance(_config['workflow'], str)
+
+    rhels_w_fips = settings.supportability.content_hosts.rhel.versions  # type [str and int]
+    rhels_no_fips = constants.DISTROS_MAJOR_VERSION.values()  # type [int]
+    if 'fips' in _config['deploy_rhel_version']:
+        assert _config['deploy_rhel_version'] in rhels_w_fips, (
+            f'{_config["deploy_rhel_version"]} not found in supported distros: \n{rhels_w_fips}'
+        )
+    else:
+        assert int(_config['deploy_rhel_version']) in rhels_no_fips, (
+            f'{_config["deploy_rhel_version"]} not found in supported distros: \n{rhels_no_fips}'
+        )
+    # Add the 3 yum repos to CV, update, needs_publish should be True
+    module_cv.repository = [repo.read() for repo in fake_yum_repos]
+    module_cv.update(['repository'])
+    module_cv = module_cv.read()
+    assert module_cv.needs_publish
+
+    # Checkout Broker host(s) via requested distro and count, or defaults.
+    with Broker(
+        **_config,
+        _count=count,
+        host_class=ContentHost,
+        deploy_network_type='ipv6' if settings.server.is_ipv6 else 'ipv4',
+    ) as brokers:
+        for chost in brokers:
```

Comment:
You could also shorten this to `chosts`, but this is more accurate to what's coming out of the context manager.
```suggestion
    ) as content_hosts:
        for chost in content_hosts:
```